documentation/src/main/asciidoc/mapstruct-reference-guide.asciidoc

```diff
@@ -924,6 +924,54 @@ public class CarMapperImpl implements CarMapper {
 ----
 ====
 
+[[passing-context]]
+=== Passing context or state objects to custom methods
+
+Additional _context_ or _state_ information can be passed through generated mapping methods to custom methods with `@Context` parameters. Such parameters are passed to other mapping methods, `@ObjectFactory` methods (see <<object-factories>>) or `@BeforeMapping` / `@AfterMapping` methods (see <<customizing-mappings-with-before-and-after>>) when applicable and can thus be used in custom code.
+
+The type of an `@Context` parameter is also inspected for factory methods and `@BeforeMapping` / `@AfterMapping` methods, which are called on the provided context parameter value if applicable.
```

Comment:
"`@Context` parameters are also searched for..."

But I'm not quite convinced of the overall idea.